File: project/hyeok-onboarding/src/main/java/com/INNER_CIRCLE_ICD4/innerCircle/domain/Question.java
package com.INNER_CIRCLE_ICD4.innerCircle.domain;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Question {

    @Id
    @GeneratedValue
    private UUID id;

    private String title;
    private String description;

    @Enumerated(EnumType.STRING)
    private QuestionType type;

    private boolean required;

    private boolean isDeleted = false;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id")
    @JsonBackReference // \uD83D\uDD01 순환 참조 방지 (Survey \u2192 Question)
    private Survey survey;

    @OneToMany(mappedBy = "question", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderColumn(name = "choice_order")
    @JsonManagedReference // \uD83D\uDD01 순환 참조 방지 (Question \u2192 Choice)
    private List<Choice> choices = new ArrayList<>();

    // 생성자
    public Question(String title, String description, QuestionType type, boolean required) {
        this.title = title;
        this.description = description;
        this.type = type;
        this.required = required;
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    // 엔티티 수정 메서드
    public void update(String title, String description, QuestionType type, boolean required) {
        this.title = title;
        this.description = description;
        this.type = type;
        this.required = required;
        this.updatedAt = LocalDateTime.now();
    }

    // Survey 연관관계 설정
    public void setSurvey(Survey survey) {
        this.survey = survey;
    }

    // 선택지 추가 메서드
    public void addChoice(Choice choice) {
        if (choices.size() >= 20) {
            throw new IllegalStateException("선택지는 최대 20개까지만 허용됩니다.");
        }
        choice.setQuestion(this);
        this.choices.add(choice);
    }

    // 테스트 및 직렬화 용도: ID 수동 설정
    public void setId(UUID id) {
        this.id = id;
    }

    // 시간 자동 설정 (JPA 라이프사이클 콜백)
    @PrePersist
    protected void onCreate() {
        LocalDateTime now = LocalDateTime.now();
        this.createdAt = now;
        this.updatedAt = now;
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: project/jinho-onboarding/src/main/kotlin/fc/innercircle/jinhoonboarding/survey/domain/Question.kt
package fc.innercircle.jinhoonboarding.survey.domain

import com.fasterxml.jackson.annotation.JsonIgnore
import fc.innercircle.jinhoonboarding.common.entity.BaseEntity
import jakarta.persistence.*
import java.util.UUID

// [설문 받을 항목]은 [항목 이름], [항목 설명], [항목 입력 형태], [항목 필수 여부]의 구성으로 이루어져있습니다.
// [항목 입력 형태]는 [단답형], [장문형], [단일 선택 리스트], [다중 선택 리스트]의 구성으로 이루어져있습니다.
// [단일 선택 리스트], [다중 선택 리스트]의 경우 선택 할 수 있는 후보를 요청 값에 포함하여야 합니다.
@Entity
class Question(
    @Id
    var id: UUID = UUID.randomUUID(),

    val title: String,
    val description: String,
    @Enumerated(EnumType.STRING)
    val questionType: QuestionType,
    val required: Boolean,
    val options: List<String> = mutableListOf(),
    var deprecated: Boolean = false,

    @ManyToOne()
    @JoinColumn(name = "survey_id", nullable = false)
    @JsonIgnore
    var survey: Survey
): BaseEntity()